analyses/chromosomal-instability/chromosomal-instability.Rmd

```diff
@@ -0,0 +1,471 @@
+      TRUE ~ "no_change"
+    ))
+  ) %>%
+  # Reformat the chromosome variable to drop the "chr"
+  dplyr::mutate(chrom = factor(gsub("chr", "", chrom),
+    levels = c(1:22, "X", "Y")
+  )) %>%
+  # Changing these so they end up matching the SV data
+  dplyr::rename(start = loc.start, end = loc.end)
+```
+
+Filter CNV data to only the changes that are larger than our cutoff `ch.pct`.
+
+```{r}
+cnv_filtered_df <- cnv_df %>%
+  dplyr::filter(changed == "change")
+```
+
+#### Set up SV data
+
+Read in the SV data.
+
+```{r}
+sv_df <- data.table::fread(file.path(data_dir, "pbta-sv-manta.tsv.gz"),
+  data.table = FALSE
+) %>%
+  # Reformat the 23 and 24 chromosomes so they are X and Y and also factors
+  dplyr::mutate(
+    chrom = dplyr::recode(SV.chrom,
+      `23` = "X", `24` = "Y"
+    )
+  )
+```
+
+#### Set up chromosome reference data
+
+Set up chromosome size data. 
+It just so happens that this BED file: `WGS.hg38.strelka2.unpadded.bed` is actually 
+just a list of the chromosome sizes so we are using that for now. 
+
+```{r}
+# Set up Chr sizes
+chr_sizes <- readr::read_tsv(file.path(data_dir, "WGS.hg38.strelka2.unpadded.bed"),
+  col_names = FALSE
+) %>%
+  # Reformat the chromosome variable to drop the "chr"
+  dplyr::mutate(X1 = factor(gsub("chr", "", X1),
+    levels = c(1:22, "X", "Y", "M")
+  )) %>%
+  # Remove sex chromosomes
+  dplyr::filter(!(X1 %in% c("X", "Y", "M")))
+
+# Make chromosome size named vector
+chr_sizes_vector <- chr_sizes$X3
+names(chr_sizes_vector) <- chr_sizes$X1
+```
+
+## Format the data as chromosomes breaks
+
+The SV and CNV data comes to us in the form of ranges, but for getting a look 
+at chromosomal instability, we will want to convert this data into single break
+points of the genome. 
+We'll also remove the sex chromosomes. 
+TODO: after updating to consensus CNV data, evaluate whether sex chromosomes should still be removed. 
+
+Make an CNV breaks data.frame. 
+
+```{r}
+# Make a CNV data.frame that has the breaks
+cnv_breaks <- data.frame(
+  samples = rep(cnv_filtered_df$ID, 2),
+  chrom = rep(cnv_filtered_df$chrom, 2),
+  coord = c(cnv_filtered_df$start, cnv_filtered_df$end),
+  seg.mean = rep(cnv_filtered_df$seg.mean, 2),
+  copy.num = rep(cnv_filtered_df$copy.num, 2),
+  stringsAsFactors = FALSE
+) %>%
+  # Remove sex chromosomes
+  dplyr::filter(!(chrom %in% c("X", "Y")))
+```
+
+Make an SV breaks data.frame. 
+
+```{r}
+# Make a data.frame that has the breaks
+sv_breaks <- data.frame(
+  samples = rep(sv_df$Kids.First.Biospecimen.ID.Tumor, 2),
+  chrom = rep(sv_df$chrom, 2),
+  coord = c(sv_df$SV.start, sv_df$SV.end),
+  svclass = rep(sv_df$SV.type, 2),
+  stringsAsFactors = FALSE
+) %>%
+  # Remove sex chromosomes and NAs
+  dplyr::filter(!(chrom %in% c("X", "Y", "M", NA)))
+```
+
+Make an common breaks data.frame. 
+
+```{r}
+common_breaks <- dplyr::bind_rows(sv_breaks, cnv_breaks) %>%
+  dplyr::distinct(samples, chrom, coord, .keep_all = TRUE)
```

Comment:
There was some discussion as to whether this should be an intersection, rather than union (which this is), but this seems correct to me. The only question I had, which we did discuss verbally, was whether we should also filter out adjacent breakpoints, where one segment starts 1 bp after the previous segment. I don't anticipate this will happen much in the consensus data, but it is a thing that could occur, and would overcount high instability regions in particular.